src/ringcentral-web-phone.js

```diff
@@ -1155,6 +1177,226 @@
+    }
+
+    /*--------------------------------------------------------------------------------------------------------------------*/
+
+    function addToMap(map, key) {
+        map = map || {};
+        if (key in map) {
+            map[key] =  parseInt(map[key]) + 1;
+        } else {
+            map[key] = parseInt(1);
+        }
+        return Object.assign({},map);
+    }
+
+    /*--------------------------------------------------------------------------------------------------------------------*/
+
+
+    //TODO: find relaible way to find network type , use navigator.connection.type?
+    function getNetworkType(connectionType){
+        var sysNetwork = connectionType.systemNetworkType || 'unknown';
+        var localNetwork = connectionType.local.networkType || ['unknown'];
+
+        if (!sysNetwork || sysNetwork === 'unknown') {
+            return localNetwork[0];
+        }
+        return sysNetwork;
+    }
+
+    /*--------------------------------------------------------------------------------------------------------------------*/
+    /**
+     * @this {SIP.Session}
+     */
+    function publishQosStats(session, qosStatsObj, options){
+        options = options || {};
+
+        var networkType = calculateNetworkUsage(qosStatsObj) || '';
+        var targetUrl = options.targetUrl || 'rtcpxr@rtcpxr.ringcentral.com:5060';
+        var event = options.event || 'vq-rtcpxr';
+        options.expires = 60;
+        options.contentType = "application/vq-rtcpxr";
+        options.extraHeaders = options.extraHeaders || [];
+        options.extraHeaders.push('p-rc-client-info:' + 'cpuRC=0:0;cpuOS=0:0;netType=' + networkType + ';ram=0:0');
+
+        var calculatedStatsObj =  calculateStats(qosStatsObj);
+        var body = createPublishBody(calculatedStatsObj);
+        var pub = session.ua.publish(targetUrl, event, body, options);
+        qosStatsObj.status = false;
+        pub.close();
+        session.emit('qos-published',body);
+    }
+
+    /*--------------------------------------------------------------------------------------------------------------------*/
+
+    /**
+     * @this {SIP.Session}
+     */
+    function startQosStatsCollection(){
+
+        var session =  this;
+        var qosStatsObj = getQoSStatsTemplate();
+
+        qosStatsObj.callID = session.request.call_id||'';
+        qosStatsObj.fromTag  = session.from_tag || '';
+        qosStatsObj.toTag  = session.to_tag || '';
+        qosStatsObj.localID = session.request.headers.From[0].raw||session.request.headers.From[0];
+        qosStatsObj.remoteID = session.request.headers.To[0].raw||session.request.headers.To[0];
+        qosStatsObj.origID = session.request.headers.From[0].raw||session.request.headers.From[0];
+
+
+        var repeatInterval = session.ua.qosCollectInterval;
+        var peer =  session.sessionDescriptionHandler.peerConnection;
+        var previousGetStatsResult;
+
+        getStats(peer, function (getStatsResult){
```

Comment:
No need for the `peer` var, use `session.sessionDescriptionHandler.peerConnection` directly